lib/inspec/dependencies/dependency_set.rb

```diff
@@ -18,14 +18,22 @@ class DependencySet
     # @param cwd [String] Current working directory for relative path includes
     # @param vendor_path [String] Path to the vendor directory
     #
-    def self.from_lockfile(lockfile, cwd, vendor_path)
+    def self.from_lockfile(lockfile, cwd, vendor_path, backend)
       vendor_index = VendorIndex.new(vendor_path)
       dep_tree = lockfile.deps.map do |dep|
-        Inspec::Requirement.from_lock_entry(dep, cwd, vendor_index)
+        Inspec::Requirement.from_lock_entry(dep, cwd, vendor_index, backend)
```

Comment:
Yeah, I agree this is gross.  The reason we started threading this through everything is that we need the backend anywhere we might instantiate a ProfileContext.  Since we instantiate ProfileContext's as we are building Inpec::Profile's, we now need it all over the place.